corehq/apps/campaign/views.py

```diff
@@ -85,16 +87,24 @@ def get_context_data(self, **kwargs):
         context.update(self.dashboard_map_case_filters_context())
         return context
 
-    @staticmethod
-    def _dashboard_gauge_configs(dashboard):
+    def _dashboard_gauge_configs(self, dashboard):
         dashboard_gauge_configs = {
             'cases': [],
             'mobile_workers': [],
         }
         for dashboard_gauge in dashboard.gauges.all():
-            dashboard_gauge_configs[dashboard_gauge.dashboard_tab].append(dashboard_gauge.to_widget())
+            dashboard_gauge_configs[dashboard_gauge.dashboard_tab].append(
+                self._get_gauge_config(dashboard_gauge)
+            )
         return dashboard_gauge_configs
 
+    @staticmethod
+    def _get_gauge_config(dashboard_gauge):
+        config = dashboard_gauge.to_widget()
+        config['value'] = get_gauge_metric_value(dashboard_gauge)
+        config['metric_name'] = dict(GAUGE_METRICS).get(dashboard_gauge.metric, '')
```

Comment:
Why would `dashboard_gauge.metric` not be in `GAUGE_METRICS`? If that ever happened, I would want to know. I'd fail loudly here.
```suggestion
        config['metric_name'] = dict(GAUGE_METRICS)[dashboard_gauge.metric]
```

Reply:
Just noting that the page actually fails in case of a metric that is not present when we compute it's value.
This would happen from here, https://github.com/dimagi/commcare-hq/pull/36096/files#diff-221d722f9d0283d18d7a0272ae1a8669a143dae982d0b196a755571eae38489aR34

Reply:
And I think that is a bad thing, so I handled that case here, 
https://github.com/dimagi/commcare-hq/pull/36096/commits/aca8e6aed3347c7b2cf51167594f6c1928f59fcc

Reply:
Also uploaded this in the related PR description, this is what a missing metric would look like.

![Screenshot from 2025-03-31 01-47-16](https://github.com/user-attachments/assets/41daf393-8583-4e4d-81e8-e57c4b6bde3d)